api/src/test/java/io/opentelemetry/trace/attributes/SemanticAttributesTest.java

```diff
@@ -51,15 +53,20 @@ public void shouldEnableSetAttributeOnSpan() throws IllegalAccessException {
         keys.add(((StringAttributeSetter) attribute).key());
         ((StringAttributeSetter) attribute).set(span, "TestValue");
         ((StringAttributeSetter) attribute).set(span, null);
+        ((StringAttributeSetter) attribute).set(builder, "TestValue");
+        ((StringAttributeSetter) attribute).set(builder, null);
       } else if (attribute instanceof LongAttributeSetter) {
         keys.add(((LongAttributeSetter) attribute).key());
         ((LongAttributeSetter) attribute).set(span, 42L);
+        ((LongAttributeSetter) attribute).set(builder, 42L);
       } else if (attribute instanceof DoubleAttributeSetter) {
         keys.add(((DoubleAttributeSetter) attribute).key());
         ((DoubleAttributeSetter) attribute).set(span, 3.14);
+        ((DoubleAttributeSetter) attribute).set(builder, 3.14);
       } else if (attribute instanceof BooleanAttributeSetter) {
         keys.add(((BooleanAttributeSetter) attribute).key());
         ((BooleanAttributeSetter) attribute).set(span, true);
+        ((BooleanAttributeSetter) attribute).set(builder, true);
```

Comment:
This test doesn't actually test anything except that you can loop over the fields of the class. Can we add some actual assertions about the functionality, rather than just making calls that don't test anything?

Reply:
Current test indeed only tested that you can call `set` method on every semantic attribute without any errors. But neither `Span` nor `Span.Builder` provides any API for reading attribute values. What else can you test here?

Reply:
You would usually mock the builder using Mockito and verify that `builder.setAttribute` was called once with `key()` and the provided value.

Reply:
You can additionally mock the Span instance and do the same.

Reply:
To tell you the truth, we don't do this at all for existing classes (in `SemanticAttributeTest` that is) and thus I feel we should do it in a follow-up PR where we apply the mocking for all of them.

Reply:
Please create an issue for it, then. We need to make sure our tests are really useful. This isn't going to be available to the instrumentation project until the next release, anyway, so I'm not sure why we're in such a hurry to get this merged.

Reply:
We have a branch against otel-java snapshots :)

Reply:
I can do verification of mock calls, no problem.

Reply:
@jkwatson take a look now, please